src/resolver.rs

```diff
@@ -392,6 +392,11 @@ fn is_file_excluded(
             break;
         }
         let settings = resolver.resolve(path, pyproject_strategy);
+        if !path.starts_with(&settings.project_root) {
+            // Bail out if we've walked past the project root
+            // (so excludes etc. are virtually "rooted" to the project).
+            break;
+        }
```

Comment:
I think something like this could work, but isn't this condition always true?

Reply:
I think we may want this, after the `match`:

```rust
        if path == settings.project_root {
            // Bail out if we've walked past the project root
            // (so excludes etc. are virtually "rooted" to the project).
            break;
        }
```

Reply:
Nope, it's not always true – `for path in path.ancestors() {` above :)

Reply:
Yep, you're right :)

Switched to this implementation + added a test that piggybacks on the existing directory structure...